### ISSUE TITLE
Failed Export

### ISSUE THREAD

**Issue by jameslashmar** [nicrivard/readwise2directory#4]
Just tried to use this. Really appreciate you putting this together by the way!

Running Windows 10 20H2, Python 3.9.1
Pip Package versions:
asgiref         3.3.1
certifi         2020.12.5
chardet         4.0.0
chromedriver    2.24.1
Django          3.1.5
idna            2.10
numpy           1.19.5
pandas          1.2.0
pathvalidate    2.3.2
pip             20.3.3
python-dateutil 2.8.1
pytz            2020.5
requests        2.25.1
selenium        3.141.0
setuptools      49.2.1
six             1.15.0
sqlparse        0.4.1
Unidecode       1.1.2
urllib3         1.26.2

So far, it spins up and logs in. Then it fails with the following.

PS C:\Users\james> python3 H:\Dropbox\Obsidian\PRISM\.obsidian\scripts\readwise2directory-master\readwise-GET.py
Importing variables from readwiseMetadata...
Checking if a valid dateFrom variable is defined in readwiseMetadata...
Checking if a valid targetDirectory variable is defined in readwiseMetadata...
Creating sourceDirectory variable from absolute path for this file...
H:\Dropbox\Obsidian\PRISM\.obsidian\scripts\readwise2directory-master directory variable defined
Loading data from JSON files in readwiseCategories to categoriesObject...
Fetching book data from readwise...
Converting readwise book data returned into JSON...
Sorting readwise book data by category...
Grouping readwise book data by category...
Appending readwise book data returned to categoriesObject...
1/11 books added or updated
2/11 books added or updated
3/11 books added or updated
4/11 books added or updated
5/11 books added or updated
6/11 books added or updated
7/11 books added or updated
8/11 books added or updated
9/11 books added or updated
10/11 books added or updated
11/11 books added or updated
Fetching highlight data from readwise...
Converting readwise highlight data returned into JSON...
Sorting readwise highlight data by category...
Grouping readwise highlight data by category...
Appending readwise highlight data returned to categoriesObject...
1/181 highlights added or updated
2/181 highlights added or updated
3/181 highlights added or updated
4/181 highlights added or updated
5/181 highlights added or updated
6/181 highlights added or updated
7/181 highlights added or updated
8/181 highlights added or updated
9/181 highlights added or updated
10/181 highlights added or updated
11/181 highlights added or updated
12/181 highlights added or updated
13/181 highlights added or updated
14/181 highlights added or updated
15/181 highlights added or updated
16/181 highlights added or updated
17/181 highlights added or updated
18/181 highlights added or updated
19/181 highlights added or updated
20/181 highlights added or updated
21/181 highlights added or updated
22/181 highlights added or updated
23/181 highlights added or updated
24/181 highlights added or updated
25/181 highlights added or updated
26/181 highlights added or updated
27/181 highlights added or updated
28/181 highlights added or updated
29/181 highlights added or updated
30/181 highlights added or updated
31/181 highlights added or updated
32/181 highlights added or updated
33/181 highlights added or updated
34/181 highlights added or updated
35/181 highlights added or updated
36/181 highlights added or updated
37/181 highlights added or updated
38/181 highlights added or updated
39/181 highlights added or updated
40/181 highlights added or updated
41/181 highlights added or updated
42/181 highlights added or updated
43/181 highlights added or updated
44/181 highlights added or updated
45/181 highlights added or updated
46/181 highlights added or updated
47/181 highlights added or updated
48/181 highlights added or updated
49/181 highlights added or updated
50/181 highlights added or updated
51/181 highlights added or updated
52/181 highlights added or updated
53/181 highlights added or updated
54/181 highlights added or updated
55/181 highlights added or updated
56/181 highlights added or updated
57/181 highlights added or updated
58/181 highlights added or updated
59/181 highlights added or updated
60/181 highlights added or updated
61/181 highlights added or updated
62/181 highlights added or updated
63/181 highlights added or updated
64/181 highlights added or updated
65/181 highlights added or updated
66/181 highlights added or updated
67/181 highlights added or updated
68/181 highlights added or updated
69/181 highlights added or updated
70/181 highlights added or updated
71/181 highlights added or updated
72/181 highlights added or updated
73/181 highlights added or updated
74/181 highlights added or updated
75/181 highlights added or updated
76/181 highlights added or updated
77/181 highlights added or updated
78/181 highlights added or updated
79/181 highlights added or updated
80/181 highlights added or updated
81/181 highlights added or updated
82/181 highlights added or updated
83/181 highlights added or updated
84/181 highlights added or updated
85/181 highlights added or updated
86/181 highlights added or updated
87/181 highlights added or updated
88/181 highlights added or updated
89/181 highlights added or updated
90/181 highlights added or updated
91/181 highlights added or updated
92/181 highlights added or updated
93/181 highlights added or updated
94/181 highlights added or updated
95/181 highlights added or updated
96/181 highlights added or updated
97/181 highlights added or updated
98/181 highlights added or updated
99/181 highlights added or updated
100/181 highlights added or updated
101/181 highlights added or updated
102/181 highlights added or updated
103/181 highlights added or updated
104/181 highlights added or updated
105/181 highlights added or updated
106/181 highlights added or updated
107/181 highlights added or updated
108/181 highlights added or updated
109/181 highlights added or updated
110/181 highlights added or updated
111/181 highlights added or updated
112/181 highlights added or updated
113/181 highlights added or updated
114/181 highlights added or updated
115/181 highlights added or updated
116/181 highlights added or updated
117/181 highlights added or updated
118/181 highlights added or updated
119/181 highlights added or updated
120/181 highlights added or updated
121/181 highlights added or updated
122/181 highlights added or updated
123/181 highlights added or updated
124/181 highlights added or updated
125/181 highlights added or updated
126/181 highlights added or updated
127/181 highlights added or updated
128/181 highlights added or updated
129/181 highlights added or updated
130/181 highlights added or updated
131/181 highlights added or updated
132/181 highlights added or updated
133/181 highlights added or updated
134/181 highlights added or updated
135/181 highlights added or updated
136/181 highlights added or updated
137/181 highlights added or updated
138/181 highlights added or updated
139/181 highlights added or updated
140/181 highlights added or updated
141/181 highlights added or updated
142/181 highlights added or updated
143/181 highlights added or updated
144/181 highlights added or updated
145/181 highlights added or updated
146/181 highlights added or updated
147/181 highlights added or updated
148/181 highlights added or updated
149/181 highlights added or updated
150/181 highlights added or updated
151/181 highlights added or updated
152/181 highlights added or updated
153/181 highlights added or updated
154/181 highlights added or updated
155/181 highlights added or updated
156/181 highlights added or updated
157/181 highlights added or updated
158/181 highlights added or updated
159/181 highlights added or updated
160/181 highlights added or updated
161/181 highlights added or updated
162/181 highlights added or updated
163/181 highlights added or updated
164/181 highlights added or updated
165/181 highlights added or updated
166/181 highlights added or updated
167/181 highlights added or updated
168/181 highlights added or updated
169/181 highlights added or updated
170/181 highlights added or updated
171/181 highlights added or updated
172/181 highlights added or updated
173/181 highlights added or updated
174/181 highlights added or updated
175/181 highlights added or updated
176/181 highlights added or updated
177/181 highlights added or updated
178/181 highlights added or updated
179/181 highlights added or updated
180/181 highlights added or updated
181/181 highlights added or updated
Appending updated highlight data to categoriesObject...
Fetching tags for 181 highlights in bulk via CSV export...
Opening new Chrome browser window...

DevTools listening on ws://127.0.0.1:59279/devtools/browser/6380f5b2-fed4-46ee-9792-f666f7d5f311
Logging into readwise using credentials provided in readwiseMetadata
Log-in successful! Redirecting to export page...
Traceback (most recent call last):
  File "H:\Dropbox\Obsidian\PRISM\.obsidian\scripts\readwise2directory-master\readwise-GET.py", line 1306, in <module>
    fetchTagsIndividuallyOrInBulk() # Function to determine whether to fetch tags individually or in bulk
  File "H:\Dropbox\Obsidian\PRISM\.obsidian\scripts\readwise2directory-master\readwise-GET.py", line 1279, in fetchTagsIndividuallyOrInBulk
    list_HighlightedAt, list_ReadwiseBookId, list_Source, list_Url, list_NumberOfHighlights, list_UpdatedAt, list_HighlightId = runFetchCsvData()
  File "H:\Dropbox\Obsidian\PRISM\.obsidian\scripts\readwise2directory-master\readwise-GET.py", line 788, in runFetchCsvData
    downloadCsvExport(downloadsDirectory, sourceDirectory, readwiseCsvExportFileName)
  File "H:\Dropbox\Obsidian\PRISM\.obsidian\scripts\readwise2directory-master\readwise-GET.py", line 599, in downloadCsvExport
    driver.find_element_by_xpath("//*[@id='MiscApp']/div/div[3]/div/div[1]/div/div[2]/div/button").click()
  File "C:\Users\james\scoop\apps\python\current\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 394, in find_element_by_xpath
    return self.find_element(by=By.XPATH, value=xpath)
  File "C:\Users\james\scoop\apps\python\current\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File "C:\Users\james\scoop\apps\python\current\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "C:\Users\james\scoop\apps\python\current\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"xpath","selector":"//*[@id='MiscApp']/div/div[3]/div/div[1]/div/div[2]/div/button"}
  (Session info: chrome=87.0.4280.141)

**Comment by nicrivard** [comment]
Hey @jameslashmar - thanks for giving the script a go and raising this issue! 

It looks like the code is running faster than the page can load which makes Selenium think the "Export CSV" button doesn't exist. 

I've added 'WebDriverWait' functions which will execute the code only after the presence of the element is located (i.e. the "Export CSV" button appears) as part of [release v1.3](https://github.com/nicrivard/readwise2directory/releases/tag/v1.3)

Let me know if you're still having issues, otherwise I will close this.

**Comment by jameslashmar** [comment]
This seems to have fixed this issue, however, it's now throwing up another.  You can close this out and i'll create a new issue. Thanks!
